assets/overlays/openstack-cinder/patches/provisioner_add_envvars.yaml

```diff
@@ -0,0 +1,13 @@
+kind: Deployment
+apiVersion: apps/v1
+spec:
+  template:
+    spec:
+      containers:
+        - name: csi-provisioner
+          env:
+            - name: ENABLE_TOPOLOGY
+              valueFrom:
+                configMapKeyRef:
+                  key: enable_topology
```

Comment:
What does this do for the csi-provisioner?

Reply:
This is the mechanism we've used to allow us to enable/disabled topology support, which is documented [here](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/cinder-csi-plugin/features.md#topology). By way of more context, the feature can only be enabled or disabled via a CLI flag (as opposed to something in the configuration file). This meant we needed a way to get the configuration from a user-defined config map (which is how we're storing the rest of our configuration) to that CLI. Using an env var was the solution we arrived in https://github.com/openshift/openstack-cinder-csi-driver-operator/pull/127. There may be better ways to do this and if so, I'm open to them, though we should probably do that separately, in a follow-up PR, in order to keep the asset diff as small as possible here.

Reply:
Topology feature is enabled by default in external-provisioner. AFAIK - this environment variable in csi-provisioner does nothing. cc @jsafrane

Reply:
We discussed this elsewhere, but the environment flag doesn't do anything by itself. Rather, it's used to provide the value to the `--feature-gates=Topology=XXX` value, as seen [here](https://github.com/openshift/csi-operator/pull/275/files#diff-fb444a9cea9026f12ee5dced1c8bc96685085720fcb3b2e307f20a09a2a7d1baR140-R147).

There is an issue here in that this feature gate is going away soon so we will need a new strategy. I'm going to work on that this cycle, but I don't think we should hold up merging this for that since this is effectively maintaining the status quo. https://github.com/openshift/csi-operator/blob/61c9fee3acd6fe92e681ac1b63dfc6dcb1d4df54/legacy/openstack-cinder-csi-driver-operator/assets/controller.yaml#L100-L115 Also, we *have* to figure out how to do this properly this cycle since we're going to lose this feature otherwise.